### PR TITLE
f/wpc-255-character-counter-count-more-blocks

### PR DESCRIPTION
**Pull request by HanaAyman**

### CODE REVIEW

**Review by bishoymelek — APPROVED** [review]
:tada:

**Review by penk85 — APPROVED** [review]

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 1]
```diff
@@ -191,17 +195,51 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+
+    const richTextBlocksCount = richTextBlocks.reduce(
+      (charCount: number, item) => charCount + countRichtextChars(0, item),
+      0
+    )
+
+    return totalTitleChars + richTextBlocksCount
+  }
+
+  function WordCounter(blocks: BlockValue[]): number {
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:229]: simple functions usually don't start with a capital letter.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 2]
```diff
@@ -191,17 +195,51 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+    return countRichtextChars(0, block.value)
+  }
+
+  function countTitleChars(block: TitleBlockListValue): number {
+    return (block.value.title as string).length + (block.value.lead as string).length
   }
 
-  function countTitle(blocks: BlockValue[]) {
+  function countQuoteChars(block: QuoteBlockListValue) {
+    return (block.value.quote as string).length + (block.value.author as string).length
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:207]: I think the `as string` should not be necessary here. I think quote and author can be `string | undefined` and you need to check first if they are undefined.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 3]
```diff
@@ -191,17 +195,51 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
-    return blocks.reduce((charLength: number, block: BlockValue) => {
-      if (!(block.type === BlockType.RichText)) return charLength
-      return countRichtextChars(charLength, block.value)
-    }, 0)
+  function countRichTextBlocksChars(block: RichTextBlockListValue) {
+    return countRichtextChars(0, block.value)
+  }
+
+  function countTitleChars(block: TitleBlockListValue): number {
+    return (block.value.title as string).length + (block.value.lead as string).length
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:203]: I think the `as string` should not be necessary here. I think quote and author can be `string | undefined` and you need to check first if they are undefined.

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 4]
```diff
@@ -325,8 +363,10 @@ export function ArticleEditor({id}: ArticleEditorProps) {
   })
 
   useEffect(() => {
-    setInfoData({charCount: countRichTextBlocksChars(blocks) + countTitle(blocks)})
-  },[isMetaDrawerOpen])
+    setInfoData({
+      charCount: 0 + WordCounter(blocks)
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleEditor.tsx:367]: why the `0 + `

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/routes/articleEditor.tsx (hunk 3 above)
Comment (HanaAyman) [reply packages/editor/src/client/routes/articleEditor.tsx]: I think the quote and author are of type string only so they are never undefined, so I don't think the check is needed.
```
export interface QuoteBlockValue {
  quote: string
  author: string
}
```

**Review by roosnic1 — APPROVED** [review]
🎉

**Review by penk85 — APPROVED** [review]

**Pull request merged**